### ISSUE TITLE
Global Menu

### ISSUE THREAD

**Issue by ASAPHAANING** [kirigayakazushin/deepin-topbar#38]
Is a global appmenu a planned feature? Something like vala panel appmenu!

Otherwise: Great work on this topbar, really works great with the distro!

**Comment by webmagnets** [comment]
I really want this.

**Comment by zahareevici** [comment]
will be great an appmenu and remove some empty space from top and autohide option
![deepinscreenshot_select-area_20180103100454](https://user-images.githubusercontent.com/23215730/34515839-5f8247c4-f074-11e7-85a3-0b70c710ac67.png)

**Comment by ASAPHAANING** [comment]
Here's an example in Deepin using Vala-panel :)

![deepinscreenshot_select-area_20180103105731](https://user-images.githubusercontent.com/6765179/34515955-fac42ce8-f074-11e7-8009-42578703c491.png)

**Comment by dimitrov-adrian** [comment]
May be integration with https://github.com/rilian-la-te/vala-panel-appmenu will be more ease to implement.

**Comment by ASAPHAANING** [comment]
Or this: https://github.com/lestcape/Gnome-Global-AppMenu

They both use the same API.

**Comment by dimitrov-adrian** [comment]
@ASAPHAANING yes and now, in fact the Gnome-Global-AppMenu extension uses unity-gtk-module which is deprecated and now supports the fork of it by the @rilian-la-te 

I have pointed this repo because, it already has integrations for Budgie, Xfce, Mate, ...etc and seems to be most updated.

Anyway, both are good example.

**Comment by rilian-la-te** [comment]
If deepin-toolbar is GTK, I can add support of vala-panel-appmenu to it.

**Comment by rilian-la-te** [comment]
This is Qt, so, it needs much more work (Qt does not support MenuModel) "

**Comment by ASAPHAANING** [comment]
@rilian-la-te Perhaps I could do some of the footwork, if you don't mind: What would I need to investigate?

**Comment by rilian-la-te** [comment]
You need to look to my repository https://github.com/rilian-la-te/menumodel-dbusmenu-translator and implement it actually (now it is just a bunch of random code).
To do it, you need some knowlege:
1. MenuModel protocol (from any GTK app with global menu)
2.How Qt draws a menu
3. GtkMenuTracker (from GTK sources).

So, you basically in that repo need to reimplement GDBusMenuModel, GDBusActionGroup and GtkMenuTracker.
